### ISSUE TITLE
Flaky tests

### ISSUE THREAD

**Issue by gianrubio** [coreos/prometheus-operator#803]
I was working on PR https://github.com/coreos/prometheus-operator/pull/784 and some tests failed. My assumption it's one test are flaky.

https://travis-ci.org/coreos/prometheus-operator/jobs/312084482

```
$ make test

?   	github.com/coreos/prometheus-operator/cmd/operator	[no test files]

?   	github.com/coreos/prometheus-operator/cmd/po-docgen	[no test files]

?   	github.com/coreos/prometheus-operator/contrib/grafana-watcher	[no test files]

?   	github.com/coreos/prometheus-operator/contrib/grafana-watcher/grafana	[no test files]

?   	github.com/coreos/prometheus-operator/contrib/grafana-watcher/updater	[no test files]

?   	github.com/coreos/prometheus-operator/contrib/prometheus-config-reloader	[no test files]

?   	github.com/coreos/prometheus-operator/example/alertmanger-webhook	[no test files]

--- FAIL: TestListOptions (0.00s)

	operator_test.go:25: LabelSelector not computed correctly

		

		Expected: "app=alertmanager,alertmanager=test"

		

		Got:      "alertmanager=test,app=alertmanager"

FAIL

FAIL	github.com/coreos/prometheus-operator/pkg/alertmanager	0.103s

?   	github.com/coreos/prometheus-operator/pkg/api	[no test files]

?   	github.com/coreos/prometheus-operator/pkg/client/monitoring	[no test files]

?   	github.com/coreos/prometheus-operator/pkg/client/monitoring/v1	[no test files]

?   	github.com/coreos/prometheus-operator/pkg/client/monitoring/v1alpha1	[no test files]

?   	github.com/coreos/prometheus-operator/pkg/k8sutil	[no test files]

--- FAIL: TestListOptions (0.00s)

	operator_test.go:25: LabelSelector not computed correctly

		

		Expected: "app=prometheus,prometheus=test"

		

		Got:      "prometheus=test,app=prometheus"

FAIL

FAIL	github.com/coreos/prometheus-operator/pkg/prometheus	8.562s

```

**Comment by brancz** [comment]
Yes I realized as well. The test is flaky because go maps don’t reflect the order key/value pairs are created in, so the order can be different. It should just be a matter of adding the other possible order as there are only two possible combinations.

**Comment by brancz** [comment]
closed by #811
